### PR TITLE
Fix markdown warnings in readmes

### PR DESCRIPTION
**Pull request by timtay-microsoft**
The transportExceptions file was accidentally deleted by me earlier, so I'm adding it back in so that the retryPolicy doc can still link to it

### CODE REVIEW

File: device/readme.md [hunk 1]
```diff
@@ -33,6 +33,6 @@ This folder contains the client library for Java.
 This folder contains various Java samples that illustrate how to use the client library.
 
 [devbox-setup]: ../doc/java-devbox-setup.md
-[run-java-sample]: ./iot-device-samples/
```
Comment (timtay-microsoft) [inline device/readme.md:36]: Links to folders should not end with a '/' character

File: doc/java-devbox-setup.md [hunk 2]
```diff
@@ -89,14 +87,12 @@ You can verify that the environment variables necessary to run **_Maven 3_** hav
 ```
 mvn --version
 ```
-<a name="installiot"></a>
```
Comment (timtay-microsoft) [inline doc/java-devbox-setup.md:92]: I'm not sure if these were done wrong, but I have seen this style of labeling in markdown before. Intellij didn't agree with them though, so I removed them and linked to the heading title instead

File: provisioning/provisioning-samples/readme.md [hunk 3]
```diff
@@ -19,8 +19,10 @@ Prepare your platform following the instructions [here][lnk-devbox-setup] to ins
 
 #### Provisioning Device Client
 
-* [Provisioning Sample for X509](provisioning-x509-sample): Shows how to register a device using X509 certificates.
-* [Provisioning Sample for TPM](provisioning-tpm-sample): Shows how to register a device using TPM with endorsement key.
+* [Provisioning Sample for X509](./provisioning-X509-sample): Shows how to register a device using X509 certificates.
+* [Provisioning Sample for TPM](./provisioning-tpm-sample): Shows how to register a device using TPM with endorsement key.
+* [Provisioning Sample for Symmetric Key Enrollment Group](./provisioning-symmetrickey-group-sample): Shows how to register a device in an enrollment group using Symmetric key authentication
+* [Provisioning Sample for Symmetric Key Individual Enrollment](./provisioning-symmetrickey-group-sample): Shows how to register a device in an individual enrollment using Symmetric key authentication
```
Comment (timtay-microsoft) [inline provisioning/provisioning-samples/readme.md:25]: I updated this readme to reflect that we now have two symmetric key samples in this folder as well

File: readme.md [hunk 4]
```diff
@@ -76,13 +76,13 @@ Devices and data sources in an IoT solution can range from a simple network-conn
...
-| X.509 Enrollment Group      | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | This SDK supports connecting your device to the Device Provisioning Service via [enrollment group](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#enrollment) using [X.509 root certificate](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-security#root-certificate).   Please visit the [samples folder](./provisioning/provisioning-samples/) to learn more about this feature.                                                                                                                                                                                            |
```
Comment (timtay-microsoft) [inline readme.md:85]: This looks like a mess to read, but all I did was remove the '/' character at the end of links that were supposed to point to folders

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**